packages/manager/apps/pci-databases-analytics/src/pages/services/create/_components/ApiTerraformDialog.component.tsx

```diff
@@ -0,0 +1,335 @@
+          // Add nodes to nodesList depending on data.nbNodes
+          for (let i = 0; i < data.nbNodes; i += 1) {
+            nodesList.push({
+              flavor: data.flavor,
+              region: data.region,
+            });
+          }
+
+          const terraformService: TTerraform.cloud.database.ServiceRequest = {
+            body: {
+              description: data.name,
+              nodesList,
+              plan: data.plan,
+              version: `${data.engineWithVersion.version}`,
+              ipRestrictions: data.ipRestrictions,
+            },
+            pathParams: {
+              engine: data.engineWithVersion
+                .engine as TTerraform.cloud.database.EngineEnum,
+              serviceName: projectData.data?.project_id,
+            },
+          };
+          if (dialogData.result.flavor.storage) {
+            terraformService.body.disk = {
+              size:
+                dialogData.result.flavor.storage.minimum.value +
+                data.additionalStorage,
+            } as Disk;
+          }
+          serviceToTerraform(terraformService);
+        },
+        (error) => {
+          console.error(error);
+          toast({
+            title: error,
+            variant: 'destructive',
+          });
+        },
+      )();
+    }
+  };
+
+  useEffect(() => {
+    // Validate form data
+    dialogData.form.handleSubmit(
+      (data) => {
+        // API
+        const apiBody: ServiceCreationWithEngine = {
+          description: data.name,
+          engine: data.engineWithVersion.engine as database.EngineEnum,
+          nodesPattern: {
+            flavor: data.flavor,
+            number: data.nbNodes,
+            region: data.region,
+          },
+          plan: data.plan,
+          version: data.engineWithVersion.version,
+          ipRestrictions: data.ipRestrictions,
+        };
+        if (data.network.type === database.NetworkTypeEnum.private) {
+          const networkOpenstackId = dialogData.result.network.network.regions.find(
+            (r) => r.region.includes(data.region),
+          ).openstackId;
+          apiBody.networkId = networkOpenstackId;
+          apiBody.subnetId = data.network.subnetId;
+        }
+        if (dialogData.result.flavor.storage) {
+          apiBody.disk = {
+            size:
+              dialogData.result.flavor.storage.minimum.value +
+              data.additionalStorage,
+          } as database.service.Disk;
+        }
+        // Destructure body to extract engine and keep the rest
+        const { engine, ...formattedApiBody } = apiBody;
+        setApiEndpoint(
+          `/cloud/project/${projectData.data?.project_id}/database/${engine}`,
+        );
+        setApiBodyContent(formattedApiBody);
+      },
+      (error) => {
+        console.error(error);
+      },
+    )();
+  }, []);
+
+  return (
+    <>
+      <Dialog open={opened} onOpenChange={onRequestClose}>
+        <DialogContent className="sm:max-w-xl">
```

Comment:
Add a ScrollArea to handle small screens
```suggestion
        <DialogContent className="px-0 sm:max-w-2xl">
            <ScrollArea className="max-h-[80vh] px-6">
```